### ISSUE TITLE
timeout with "error event from" error message

### ISSUE THREAD

**Issue by rajatjindal** [helm/helm#8512]
Output of `helm version`:  fork of helm 2.16.x (we have some very minor changes in addition to original helm 2.16)

Output of `kubectl version`: 

server: 1.16.11
client: 1.16.x

Cloud Provider/Platform (AKS, GKE, Minikube etc.): AWS (ELB in front of api server with 1hour idle timeout)

We have a pipeline that takes about 90m to rollout (due to rollout strategy to replace one pod at a time), and we started to see failures in the pipeline with error msg "error event from" after about 45 minutes of post-install webhook running.

**Comment by rajatjindal** [comment]
We tried a few things to ensure its not coming from post install job hook itself, and during that process we observed that the job itself continues to run in the background but tiller gives the said error and cause pipeline to fail.

on further debugging, we stumbled upon this discussion between helm maintainers https://github.com/helm/helm/pull/4734#discussion_r222795375 

that helped us found this PR https://github.com/kubernetes/kubernetes/pull/66971 which added a new function `UntilWithSync` that was suppose to fix this issue. and also found this PR: https://github.com/kubernetes/kubernetes/pull/67817 which fixes similar issue in `kubectl rollout status` command.

we added this fix into our fork of tiller/helm (I just want to reiterate, which is very very close to 2.16.x branch of helm/helm), and haven't seen that issue after this. (last time it took 77 minutes for the pipeline to run). and we had a few successful runs on that particular project and all other projects using updated helm/tiller.
